ouroboros-consensus/src/Ouroboros/Consensus/Protocol/PBFT.hs

```diff
@@ -343,6 +343,27 @@ instance PBftCrypto c => ConsensusProtocol (PBft c) where
     where
       params = pbftWindowParams cfg
 
+  reupdateChainDepState cfg@PBftConfig{..}
+                        toValidate
+                        slot
+                        (TickedPBftState (TickedPBftLedgerView dms) state) =
+      case toValidate of
+        PBftValidateBoundary hash ->
+          appendEBB cfg params slot hash state
+        PBftValidateRegular PBftFields{pbftIssuer} _ _ ->
+          case Bimap.lookupR (hashVerKey pbftIssuer) dms of
+            Nothing ->
+              error $ show $ PBftNotGenesisDelegate
```

Comment:
I'm not too happy about these `error . show` combos I have here and in other places, but we basically had the same `error` in `cantBeError` (in `Extended.hs`).

I do like the approach taken in `cardano-ledger-specs`, i.e., skipping specific checks (we would need to introduce some kind of predicate for this) can be done by flipping a switch.